internal/controller/runtime/fsm/runtime_fsm_patch_shoot_test.go

```diff
@@ -0,0 +1,86 @@
+package fsm
+
+import (
+	"context"
+	"time"
+
+	gardener "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	"github.com/kyma-project/infrastructure-manager/internal/controller/metrics/mocks"
+	. "github.com/onsi/ginkgo/v2" //nolint:revive
+	. "github.com/onsi/gomega"    //nolint:revive
+	"github.com/onsi/gomega/types"
+	"github.com/stretchr/testify/mock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	util "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/utils/ptr"
+)
+
+var _ = Describe("KIM sFnPatchExistingShoot", func() {
+
+	testCtx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	testScheme := runtime.NewScheme()
+	util.Must(imv1.AddToScheme(testScheme))
+	util.Must(gardener.AddToScheme(testScheme))
+
+	withMockedMetrics := func() fakeFSMOpt {
+		m := &mocks.Metrics{}
+		m.On("SetRuntimeStates", mock.Anything).Return()
+		m.On("CleanUpRuntimeGauge", mock.Anything, mock.Anything).Return()
+		m.On("IncRuntimeFSMStopCounter").Return()
+		return withMetrics(m)
+	}
+
+	inputRtWithForceAnnotation := makeInputRuntimeWithAnnotation(map[string]string{"operator.kyma-project.io/force-patch-reconciliation": "true"})
+
+	testShoot := gardener.Shoot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-shoot",
+			Namespace: "garden-",
+		},
+		Spec: gardener.ShootSpec{
+			DNS: &gardener.DNS{
+				Domain: ptr.To("test-domain"),
+			},
+		},
+		Status: gardener.ShootStatus{
+			LastOperation: &gardener.LastOperation{
+				State:          gardener.LastOperationStateSucceeded,
+			},
+		},
+	}
+
+	testFunction := buildPatchTestFunction(sFnPatchExistingShoot)
+
+	DescribeTable(
+		"transition graph validation for sFnPatchExistingShoot",
+		testFunction,
+		Entry(
+			"should update status after succesful patching and remove force patch annotation",
+			testCtx,
+			must(newFakeFSM, withMockedMetrics(), withTestFinalizer,  withFakedK8sClient(testScheme, inputRtWithForceAnnotation), withFakeEventRecorder(1)),
+			&systemState{instance: *inputRtWithForceAnnotation, shoot: &testShoot},
+			haveName("sFnUpdateStatus"),
+			nil,
```

Comment:
I assume that `nil` is "expectedAnnotation". I think we should make it more clear what this parameter is for. How about changing this into: 

```
	var expectedAnnotations map[string]string


Entry(
			"should update status after succesful patching and remove force patch annotation",
			testCtx,
			must(newFakeFSM, withMockedMetrics(), withTestFinalizer, withFakedK8sClient(testScheme, inputRtWithForceAnnotation), withFakeEventRecorder(1)),
			&systemState{instance: *inputRtWithForceAnnotation, shoot: &testShoot},
			haveName("sFnUpdateStatus"),
			expectedAnnotations,
		),
```